scripts/abi_artifact_generator/config/architectures.yml

```diff
@@ -1,98 +1,126 @@
 ---
 "x86_64":
   register_type: uint64_t
+  intel_register_type: qword
   register_size: 8
   register_list: [rax, rbx, rcx, rdx, rsi, rdi, r8, r9, r10, r11, r12, r13, r14, r15]
+  styles:
```

Comment:
1. `comment_styles`?
2. Is this the only relevant difference so far between the GNU and the MSVC assembler?

Reply:
I really thought about that, but both ways sound about equally complex. Except having the outputs side-by-side made debugging them a lot easier (since you see both renderings of the things at once), kind of like here:
```cpp
;; .global execute_single_huge_struct_argument
;; execute_single_huge_struct_argument:
execute_single_huge_struct_argument proc
  { big piece of assembly omitted because it's the same for both }
;; #ifndef CalleeIsResponsibleForStackCleanup
  add esp, 512
;; #endif
  mov eax, dword ptr [saved_return_address]
  push eax
  ret
execute_single_huge_struct_argument endp
```

Even while on msvc version you can still see the equivalent gcc code.

Reply:
But I guess if you'd rather the python-side harbored the extra complexity, I'm not too opposed.

Reply:
```

/*( gcc )*/#include "../abi.h"

/*( gcc )*/#if defined(TARGET_x86_64) || defined(TARGET_i386)
/*( gcc )*/.intel_syntax noprefix
/*( gcc )*/#endif

/*( msvc )*/ifndef rax
/*( msvc )*/.model flat, c
/*( msvc )*/endif

/*( msvc )*/extern saved_return_address: /*( intel_register_type )*/
/*( msvc )*/extern randomized_state: /*( intel_register_type )*/

/*( msvc )*/extern regenerate_state: proc
/*( msvc )*/extern print_current_state: proc
```

vs

```
/* if gcc */
#include "../abi.h"

#if defined(TARGET_x86_64) || defined(TARGET_i386)
.intel_syntax noprefix
#endif
/* elif msvc: */
ifndef rax
.model flat, c
endif

extern saved_return_address: /*( intel_register_type )*/
extern randomized_state: /*( intel_register_type )*/

extern regenerate_state: proc
extern print_current_state: proc
/* endif */
```

Second seems much better to me, plus no polluted output.
It's more debatable on single line situations, but still, let's go with the `if`s.

Reply:
Sounds good.

Reply:
Updates?

Reply:
Pushed it. Not sure if I prefer the new approach from readability standpoint though